test/test/boot.js

```diff
@@ -404,6 +404,119 @@ function configureJasmineEnvironment() {
+ * top-level, and an inner frame of Karma that is raw HTML as text with no
+ * origin.
+ *
+ * With all of these complexities, the only way to access cast.__platform__ is
+ * asynchronously via postMessage.  This means all callers of
+ * cast.__platform__.canDisplayType must use `await`, even though the
+ * underlying method is synchronous.  If the caller is running in the top frame
+ * (as in a real receiver), `await` will do no harm.  If the caller is running
+ * inside our tests in Karma, the `await` is critical to access __platform__
+ * via this shim.
+ */
+function patchCastCanDisplayType() {
+  if (!navigator.userAgent.includes('CrKey')) {
+    // Not Chromecast.  Do nothing.
+    return;
+  }
+
+  // Create the namespaces if needed.
+  if (!window.cast) {
+    window.cast = {};
+  }
+  if (!cast.__platform__) {
+    cast.__platform__ = {};
+  }
+
+  if (cast.__platform__.canDisplayType) {
+    // Already exists, do nothing.
+    return;
+  }
+
+  // Create an async shim.  Calls to canDisplayType will be translated into
+  // async messages to the top frame, which will then execute the method and
+  // post a message back with results (or an error).  The resolve/reject
+  // functions for the shim's returned Promise will be stored temporarily in
+  // these maps and matched up by request ID.
+  /** @type {!Map<number, function()>} */
+  const resolveMap = new Map();
+  /** @type {!Map<number, function()>} */
+  const rejectMap = new Map();
+  /** @type {number} */
+  let nextId = 0;
+
+  // Listen for message events for results/errors from the top frame.
+  window.addEventListener('message', (event) => {
+    const data = event.data;
+    console.log('Received cross-frame message', data);
+
+    if (data.type == 'cast.__platform__:result') {
+      // Find the matching resolve function and resolve the promise for this
+      // request.
+      const resolve = resolveMap.get(data.id);
+      if (resolve) {
+        resolve(data.result);
+
+        // Clear both resolve and reject from the maps for this ID.
+        resolveMap.delete(data.id);
+        rejectMap.delete(data.id);
+      }
+    } else if (data.type == 'cast.__platform__:error') {
+      // Find the matching reject function and reject the promise for this
+      // request.
+      const reject = rejectMap.get(data.id);
+      if (reject) {
+        reject(data.result);
+
+        // Clear both resolve and reject from the maps for this ID.
+        resolveMap.delete(data.id);
+        rejectMap.delete(data.id);
+      }
+    }
+  });
+
+  // Shim canDisplayType to proxy the request up to the top frame.
+  cast.__platform__.canDisplayType = function() {
+    return new Promise((resolve, reject) => {
+      // Craft a message for the top frame to execute this method for us.
+      const message = {
+        id: nextId++,
+        type: 'cast.__platform__',
+        command: 'canDisplayType',
+        args: Array.from(arguments),
```

Comment:
nit; MDN suggests that `...rest` params should be preferred in a modern code. Also, I think you won't need to convert them to Array.

Reply:
Replaced with an explicit function argument, since there is only one.